Makefile

```diff
@@ -138,49 +152,65 @@ multi-arch-image-%: binary-linux-amd64 binary-linux-arm64
 
 ## Clean up static directory and build a Docker Traefik image
 .PHONY: build-image
+#? build-image: Clean up static directory and build a Docker Traefik image
 build-image: export DOCKER_BUILDX_ARGS := --load
 build-image: export DOCKER_BUILD_PLATFORMS := linux/$(GOARCH)
 build-image: clean-webui
 	@$(MAKE) multi-arch-image-latest
 
 ## Build a Docker Traefik image without re-building the webui when it's already built
 .PHONY: build-image-dirty
+#? build-image-dirty: Build a Docker Traefik image without re-building the webui when it's already built
 build-image-dirty: export DOCKER_BUILDX_ARGS := --load
 build-image-dirty: export DOCKER_BUILD_PLATFORMS := linux/$(GOARCH)
 build-image-dirty:
 	@$(MAKE) multi-arch-image-latest
 
 ## Build documentation site
 .PHONY: docs
+#? docs: Build documentation site
 docs:
 	make -C ./docs docs
 
 ## Serve the documentation site locally
 .PHONY: docs-serve
+#? docs-serve: Serve the documentation site locally
 docs-serve:
 	make -C ./docs docs-serve
 
 ## Pull image for doc building
 .PHONY: docs-pull-images
+#? docs-pull-images: Pull image for doc building
 docs-pull-images:
 	make -C ./docs docs-pull-images
 
 ## Generate CRD clientset and CRD manifests
 .PHONY: generate-crd
+#? generate-crd: Generate CRD clientset and CRD manifests
 generate-crd:
 	@$(CURDIR)/script/code-gen-docker.sh
 
 ## Generate code from dynamic configuration https://github.com/traefik/genconf
 .PHONY: generate-genconf
+#? generate-genconf: Generate code from dynamic configuration github.com/traefik/genconf
 generate-genconf:
 	go run ./cmd/internal/gen/
 
 ## Create packages for the release
 .PHONY: release-packages
+#? release-packages: Format the Code
 release-packages: generate-webui
 	$(CURDIR)/script/release-packages.sh
 
 ## Format the Code
 .PHONY: fmt
+#? fmt: Format the Code
 fmt:
 	gofmt -s -l -w $(SRCS)
+
+#? help: Get more info on make commands
+help: Makefile
+	@echo " Choose a command run in traefik:"
+	@sed -n 's/^#?//p' $< | column -t -s ':' |  sort | sed -e 's/^/ /'
```

Comment:
```suggestion
	@sed -n 's/^##//p' $< | column -t -s ':' |  sort | sed -e 's/^/ /'
```

Why not using the comment already in place?

Reply:
In my opinion, `#?` is more particular than `##`, and won't be effected by `###+`

Reply:
What is the meaning of `###+` in a Makefile?

As I can read [here](https://www.gnu.org/software/make/manual/make.html#Makefile-Contents):

> ‘#’ in a line of a makefile starts a comment. It and the rest of the line are ignored, except that a trailing backslash not escaped by another backslash will continue the comment across multiple lines. A line containing just a comment (with perhaps spaces before it) is effectively blank, and is ignored. If you want a literal #, escape it with a backslash (e.g., \#). Comments may appear on any line in the makefile, although they are treated specially in certain situations.

To me the maintain of the Makefile will be more difficult if we introduce a new format just for the help target

Reply:
Sorry about this. 

I used `###+` as a regex pattern, which means that there may be more than two `#` in the Makefile.

The reason why I use `#?` is to avoid someone using `###` for a comment or something to describe a target group (see the following screenshot), or something else.
![image](https://github.com/traefik/traefik/assets/25278203/32412cc7-4b66-4b5c-8162-56debcefb09d)

Reply:
No worries, thanks for the explanation.

To me we should get rid of `##` comment and only kept `#?`